documentation/modules/cruise-control/con-rebalance-performance.adoc

```diff
@@ -9,6 +9,33 @@
 You can adjust several performance tuning options for cluster rebalances.
 These options control how partition replica and leadership movements in a rebalance are executed, as well as the bandwidth that is allocated to a rebalance operation.
 
+== Rebalancing modes
```

Comment:
Rebalance modes are not really a performance tuning option? They are a level above this. I think this would actually be better as a whole new section below "Why use cruise control" that explains the different types of rebalance that are possible. WDYT @PaulRMellor?

Reply:
Also looking through the whole CC section in the docs we may need to clarify how each section changes (if at all) with respect to the full/add/remove.

Reply:
> Rebalance modes are not really a performance tuning option? They are a level above this. I think this would actually be better as a whole new section below "Why use cruise control" that explains the different types of rebalance that are possible.

I had a chat with @PaulRMellor yesterday and this section should be a more general concepts section. I would agree that the title about performance tuning option is not the best. Maybe we should just try to change it?

> Also looking through the whole CC section in the docs we may need to clarify how each section is effected (if at all) with respect to the full/add/remove.

We went through all sections and I didn't find anything affected unless the "Generating optimization proposals" with the procedure. We thought that the best way was not having three different procedure with a lot of duplication but relying on the single one we have today. The first step just explains how to use the 3 modes. The other steps are actually the same for all the 3 modes.

Reply:
I can take a look at this when I do an editorial review

Reply:
About the title ... "scaling" makes sense only with the two new modes and not with the "full" one. I think it's misleading to be there.
Fine about where positioning the section.